app/scripts/analysis/tools/DimensionReduction/DimensionReductionMainCtrl.ctrl.coffee

```diff
@@ -1,14 +1,15 @@
 'use strict'
 
 BaseCtrl = require 'scripts/BaseClasses/BaseController.coffee'
+DimensionReductionSidebarCtrl = require 'scripts/analysis/tools/DimensionReduction/DimensionReductionSidebarCtrl.ctrl.coffee'
```

Comment:
We should not inject Controllers into other Controllers. Instead, we should use Angular's broadcasting mechanism for data exchange between controllers. For example, see the usage of [`$on`](https://github.com/SOCR/SOCRAT/blob/f71f749fb6d69bd8780bcc1dd3407e9c40b3c0c7/app/scripts/analysis/tools/Cluster/ClusterMainCtrl.ctrl.coffee#L27) and [`$broadcast`](https://github.com/SOCR/SOCRAT/blob/f71f749fb6d69bd8780bcc1dd3407e9c40b3c0c7/app/scripts/analysis/tools/Cluster/ClusterSidebarCtrl.ctrl.coffee#L78) in the Cluster module. 
Another option is to abstract out the dataset list into a separate Angular service that can be injected in both controllers and provide a method for getting the list, e.g., see how the list of clustering algorithms is defined in [a separate service](https://github.com/SOCR/SOCRAT/blob/f71f749fb6d69bd8780bcc1dd3407e9c40b3c0c7/app/scripts/analysis/tools/Cluster/ClusterAlgorithms.service.coffee#L20) and then how the SibebarCtrl can get this list by calling [a method](https://github.com/SOCR/SOCRAT/blob/f71f749fb6d69bd8780bcc1dd3407e9c40b3c0c7/app/scripts/analysis/tools/Cluster/ClusterSidebarCtrl.ctrl.coffee#L16) of the injected service.